internal/characters/kirara/skill.go

```diff
@@ -0,0 +1,205 @@
+		return
+	}
+	c.AddStatus(kickParticleICDKey, 0.5*60, true)
+	c.Core.QueueParticle("kirara-kick", 3, attributes.Dendro, c.ParticleDelay)
+}
+
+func (c *char) rollParticleCB(a combat.AttackCB) {
+	if a.Target.Type() != targets.TargettableEnemy {
+		return
+	}
+	if c.StatusIsActive(rollParticleICDKey) {
+		return
+	}
+	c.AddStatus(rollParticleICDKey, 4*60, true)
+	c.Core.QueueParticle("kirara-roll", 1, attributes.Dendro, c.ParticleDelay)
+}
+
+func (c *char) skillPress(p map[string]int) action.ActionInfo {
+	ai := combat.AttackInfo{
+		ActorIndex:         c.Index,
+		Abil:               "Tail-Flicking Flying Kick",
+		AttackTag:          attacks.AttackTagElementalArt,
+		ICDTag:             attacks.ICDTagNone,
+		ICDGroup:           attacks.ICDGroupDefault,
+		StrikeType:         attacks.StrikeTypeSlash,
+		Element:            attributes.Dendro,
+		Durability:         25,
+		Mult:               skillPress[c.TalentLvlSkill()],
+		HitlagHaltFrames:   0.06 * 60,
+		HitlagFactor:       0.01,
+		CanBeDefenseHalted: true,
+	}
+
+	c.Core.QueueAttack(
+		ai,
+		combat.NewCircleHitOnTargetFanAngle(c.Core.Combat.Player(), geometry.Point{Y: 1}, 3, 270),
+		skillPressHitmark,
+		skillPressHitmark,
+		c.kickParticleCB,
+	)
+
+	if c.Base.Cons >= 6 {
+		c.c6()
+	}
+
+	c.QueueCharTask(c.generateSkillShield, 7*2)
+	c.SetCDWithDelay(action.ActionSkill, 8*60, 7*2)
+
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(skillPressFrames),
+		AnimationLength: skillPressFrames[action.InvalidAction],
+		CanQueueAfter:   skillPressFrames[action.ActionDash],
+		State:           action.SkillState,
+	}
+}
+
+func (c *char) skillHold(p map[string]int, duration int) action.ActionInfo {
+	// ticks
+	d := c.createSkillHoldSnapshot()
+
+	for i := 0; i <= duration; i += 20 * 2 { // 1 tick for sure
+		c.Core.Tasks.Add(func() {
+			// pattern shouldn't snapshot on attack event creation because the skill follows the player
+			d.Pattern = combat.NewCircleHitOnTarget(c.Core.Combat.Player(), nil, 3)
+			c.Core.QueueAttackEvent(d, 0)
+		}, 18+i)
+	}
+
+	// Flipclaw Strike DMG
+	ai := combat.AttackInfo{
+		ActorIndex:       c.Index,
+		Abil:             "Flipclaw Strike",
+		AttackTag:        attacks.AttackTagElementalArt,
+		ICDTag:           attacks.ICDTagNone,
+		ICDGroup:         attacks.ICDGroupDefault,
+		StrikeType:       attacks.StrikeTypeDefault,
+		Element:          attributes.Dendro,
+		Durability:       25,
+		Mult:             flipclawDmg[c.TalentLvlSkill()],
+		HitlagHaltFrames: 0.02 * 60,
+		HitlagFactor:     0.05,
+	}
+	snap := c.Snapshot(&ai)
+	c.Core.QueueAttackWithSnap(
+		ai,
+		snap,
+		combat.NewCircleHitOnTarget(c.Core.Combat.Player(), geometry.Point{Y: 0.5}, 3),
+		(skillHoldKickHitmark-600)+duration,
+		c.kickParticleCB,
+	)
+
+	if c.Base.Cons >= 6 {
+		c.c6()
+	}
+
+	c.QueueCharTask(c.generateSkillShield, 7*2)
+	c.SetCDWithDelay(action.ActionSkill, int(8*60+float64(duration)*0.4), (skillHoldCDStart-600)+duration)
```

Comment:
The cooldown isn't 100% linear like this. It should be more like `cd := 8*60+duration/30*12` so that every 0.5s of Neko Parcel state, another 0.2s of cooldown are added.